lib/doodbalib/__init__.py

```diff
@@ -113,6 +113,8 @@ def addons_config(filtered=True, strict=False):
     logger.debug("Merged addons definition before expanding: %r", all_globs)
     # Expand all globs and store config
     for repo, partial_globs in all_globs.items():
+        to_exclude = [x[1:] for x in
+                      filter(lambda g: g.startswith('!'), partial_globs)]
```

Comment:
It seems to me like this will not work with glob patterns. :thinking:  Please test that use case (see comments below) and make it pass. :blush:

Reply:
this is using the strings that were feched before from the addons.yml file and extracting the lines starting with `!` to use it afterwards, it does not modify the result of `found = glob(full_glob)`

so it is not expecting the result of the glob to change but to store in a list the modules to exclude later simple with this:
```python
                if addon in to_exclude:
                    logger.debug("Excluded '%s'", addon)
                    continue
```

So it is not relying in the glob functionalities but doing a work-around.

Reply:
I think it would be better to do something like this pseudo-code:

```python
matches = {"include": set(), "exclude": set()}
for glob in globs:
  # Know if the glob is to include or exclude
  mode = "exclude" if glob.startswith("!") else "include"
  if mode == "exclude":
    # Remove "!" from the start
    glob = glob[1:]
  matched_dirs = expand_glob(glob)
  matches[mode] |= matched_dirs
# The final list of addons to include removes the excluded ones
winners = matches["include"] - matches["exclude"]
```